core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -2788,38 +2788,50 @@ public static double power(BigDecimal b0, BigDecimal b1) {
 
   // LN, LOG, LOG10, LOG2
 
-  /** SQL {@code LOG(number, number2)} function applied to double values. */
-  public static @Nullable Double log(double number, double number2, int nullFlag) {
-    if (nullFlag == 1 && number <= 0) {
+  /** SQL {@code LOG(number, base)} function applied to double values. */
```

Comment:
The meaning of 'flag' is very hard to follow from the call to the callee.
I think that a better way is to really have different functions, corresponding to the 3 different versions of 'flag'. Each function can have a nice descriptive name, like log_base_value

Reply:
I'm worried that this makes the code redundant, so I thought I could add a comment to make this clear.